ext/srt/gstsrtobject.c

```diff
@@ -829,6 +851,48 @@ thread_func (gpointer data)
   }
 }
 
+static GSocketAddress *
+peeraddr_to_g_socket_address (const struct sockaddr *peeraddr) {
+  gsize peeraddr_len;
+
+  switch (peeraddr->sa_family) {
+    case AF_INET:
+      peeraddr_len = sizeof (struct sockaddr_in);
+      break;
+    case AF_INET6:
+      peeraddr_len = sizeof (struct sockaddr_in6);
+      break;
+    default:
+      g_warning ("Unsupported address family %d", peeraddr->sa_family);
+      return NULL;
+  }
+  return g_socket_address_new_from_native ((gpointer) peeraddr, peeraddr_len);
+}
+
+static gint
+srt_listen_callback_func(GstSRTObject * self, SRTSOCKET sock, int hs_version,
+    const struct sockaddr* peeraddr, const char* stream_id) {
+  g_autoptr (GSocketAddress) addr = peeraddr_to_g_socket_address (peeraddr);
+
+  if (self->authentication) {
+    gboolean authenticated = FALSE;
+
+    /* notifying caller-connecting */
+    g_signal_emit_by_name (self->element, "caller-connecting", addr,
+      stream_id, &authenticated);
+
+    if (!authenticated)
+      goto reject;
+  }
+
+  g_debug ("Accepting sink %d streamid: %s", sock, stream_id);
+  return 0;
+reject:
+  /* notifying caller-rejected */
+  g_signal_emit_by_name (self->element, "caller-rejected");
```

Comment:
I'd add some information that identify the rejected called into the callback. Let's put `addr` and `stream_id` as parameters, same as `"caller-connecting"`.